atomix/cluster/src/main/java/io/atomix/raft/protocol/ReplicatableJournalRecord.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © 2020 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atomix.raft.protocol;
+
+public final class ReplicatableJournalRecord implements ReplicatableRecord {
```

Comment:
:wrench: Could be a record. Better do it now before a release, as changing from class to record in Kryo is a non-backwards compatible change.

```java
public record ReplicatableJournalRecord(
    long term, long index, long checksum, byte[] serializedJournalRecord)
    implements ReplicatableRecord {

  /**
   * Returns the approximate size needed when serializing this class. The exact size depends on the
   * serializer.
   *
   * @return approximate size
   */
  public int approximateSize() {
    return serializedJournalRecord.length + Long.BYTES + Long.BYTES + Long.BYTES;
  }
}
```